aip/general/0130.md

```diff
@@ -0,0 +1,77 @@
+| Custom collection mutation methods | Perform an imperative operation referencing a collection that may mutate one or more resources within that collection in fashion that cannot be easily achieve by standard methods (e.g. state transitions). | [AIP-136][]                                        | handwritten                     | automatable     | automatable    | automatable     |
+| **Resource-oriented**              |                                                                                                                                                                                                              |                                                    |                                 |                 |                |                 |
+| Standard resource methods          | Fetch or mutate a single resource (Get, Update, Delete).                                                                                                                                                     | [AIP-121][], [AIP-131][], [AIP-134][], [AIP-135][] | automatable                     | automatable     | automatable    | automatable     |
+| Custom resource fetch methods      | Fetch information across a single resource that cannot be expressed via a standard method.                                                                                                                   | [AIP-136][]                                        | automatable                     | automatable     | automatable    | automatable     |
+| Custom resource mutation methods   | Perform an imperative operation on a resource that may mutate it in a way a standard method cannot (e.g. state transitions).                                                                                 | [AIP-136][]                                        | handwritten                     | automatable     | automatable    | automatable     |
+| **None of the above**              |                                                                                                                                                                                                              |                                                    |                                 |                 |                |                 |
+| Streaming methods                  | Methods that communicate bi-directionally.                                                                                                                                                                   |                                                    | handwritten                     | handwritten     | handwritten    | automatable     |
```

Comment:
I don't think `Streaming methods` should be limited to bi-directional, because there is `client-stream` and `server-stream` as well. Folks will get hung up on that if they have proto/grpc experience.

Reply:
ack. Do you have a suggested way to state that? we could be general and say "methods that communicate via streaming" but that definition feels a little cyclic.